### PR TITLE
Enable forward compatibility

### PR DESCRIPTION
**Pull request by bjoernthiel**
OBF is backward compatible, so the reader should be forward compatible.

### CODE REVIEW

**Comment by sbesson** [comment]
Hi @bjoernthiel, many thanks for the contribution. Could you expand a bit on the PR description to describe how to test the patch? In particular, do you have any corresponding sample files that we would also add to our data repository for be daily checked using our non-regression tests?

On a technical level, we are trying to avoid removing/adding private fields to the readers in patch releases to minimize serialization breakages. Could the two dropped `private static final` fields be marked as deprecated instead? We will clean them in the following minor version increment of Bio-Formats.

**Comment by bjoernthiel** [comment]
The stack version and possibly the file version as well are are going to be increased in a future version of the OBF file format. Unfortunately I don't have example data yet. But as OBF is backward compatible these version checks are useless. 

Dropping the corresponding fields later is OK with me.

**Comment by dgault** [comment]
Marking for inclusion in daily builds and tests
--include

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

File: components/formats-bsd/src/loci/formats/in/OBFReader.java [hunk 1]
```diff
@@ -135,9 +132,9 @@ private int getFileVersion(RandomAccessInputStream stream) throws IOException
   @Override
   public boolean isThisType(RandomAccessInputStream stream) throws IOException
   {
-    final int fileVersion = getFileVersion(stream);
+    final int fileVersion = getFileVersion(stream) ;
```
Comment (sbesson) [inline components/formats-bsd/src/loci/formats/in/OBFReader.java]: Remove extra whitespace before the semicolon.

File: components/formats-bsd/src/loci/formats/in/OBFReader.java [hunk 2]
```diff
@@ -135,9 +132,9 @@ private int getFileVersion(RandomAccessInputStream stream) throws IOException
...
 
-    return fileVersion >= 0 && fileVersion <= FILE_VERSION;
+    return fileVersion >= 0 ;
```
Comment (sbesson) [inline components/formats-bsd/src/loci/formats/in/OBFReader.java]: Remove extra whitespace before the semi-colon

**Pull request merged**